compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -175,6 +176,9 @@ public Symbol getBCompiledSymbol(BSymbol symbol, String name) {
             if (symbol.owner instanceof BObjectTypeSymbol) {
                 return createObjectFieldSymbol((BVarSymbol) symbol);
             }
+            if (symbol.owner instanceof BTypeSymbol && Symbols.isFlagOn(symbol.flags, Flags.FIELD)) {
```

Comment:
Why do we need the check for `Symbols.isFlagOn(symbol.flags, Flags.FIELD)` only with tuples?

Reply:
That's the only way to check whether the particular symbol is a tuple member or not

Reply:
My concern was why specifically for tuples, given that we haven't done so for record, object, class, etc.

Reply:
Others are captured according to the type such as `symbol.owner instanceof BObjectTypeSymbol`, symbol.owner instanceof BRecordTypeSymbol`. But here we don't have a one as `BTupleTypeSymbol` or any relatable type. We just get it as `BTypeSymbol`

Reply:
When we get to this point, can it be anything other than a tuple?

Btw, we can probably add a `BTupleTypeSymbol` also, if it will be useful. @ushirask, let's create an issue for this.

Reply:
Created in [#39185](https://github.com/ballerina-platform/ballerina-lang/issues/39185)

Reply:
>When we get to this point, can it be anything other than a tuple?
According to the current implementation of the compiler, it can't be anything other than a tuple here (L179). But what if we introduce something new?

> Btw, we can probably add a BTupleTypeSymbol also, if it will be useful. @ushirask, let's create an issue for this.
+1

Reply:
@MaryamZi Are we going to keep this as it is till https://github.com/ballerina-platform/ballerina-lang/issues/39185 gets fixed?

Reply:
Let's keep it as is. I don't have a strong opinion about the requirement for a BTupleTypeSymbol yet, we need to look into it more, so may or may not get added.

Reply:
Ok. IMO, having a `TupleTypeSymbol` will not change anything soon, but it will be clean and beneficial (in future)